src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,21 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (storageDao.get(user.getLogin()).equals(null)) {
+            throw new RuntimeException("Login is exist");
+        }
+        if (user.getLogin().equals(null) || user.getPassword().equals(null)) {
+            throw new RuntimeException("Your password or login is incorrect");
```

Comment:
Checking for null with `.equals(null)` is incorrect; use `==` instead. The exception message is not informative enough. Split this condition into separate checks for login and password to provide more specific feedback to the user.